app/page/importToken/index.tsx

```diff
@@ -0,0 +1,129 @@
+import { useContext, useState } from "react"
+import { useHashLocation } from "wouter/use-hash-location"
+
+import { Button } from "~app/component/button"
+import { Divider } from "~app/component/divider"
+import { Input } from "~app/component/input"
+import { StepBackHeader } from "~app/component/stepBackHeader"
+import { ProviderContext } from "~app/context/provider"
+import { ToastContext } from "~app/context/toastContext"
+import { useAccount, useAction, useTokens } from "~app/hook/storage"
+import { Path } from "~app/path"
+import { getErc20Contract } from "~packages/network/util"
+import address from "~packages/util/address"
+import number from "~packages/util/number"
+
+export function ImportToken() {
+  const [, navigate] = useHashLocation()
+  const { provider } = useContext(ProviderContext)
+  const { setToast } = useContext(ToastContext)
+
+  const { importToken } = useAction()
+
+  const account = useAccount()
+  const tokens = useTokens()
+
+  const [tokenAddress, setTokenAddress] = useState("")
+  const [tokenAddressValid, setTokenAddressValid] = useState(false)
```

Comment:
I think `tokenAddressValid` is a derived state variable, which can be calculated by token info (checking if there are symbol, token decimals)

Reply:
Refined in [8b7ac9c](https://github.com/pilagod/waallet-extension/pull/178/commits/8b7ac9c811510bf92f74122023ffdcae49a7198f)